keras_cv/models/segmentation/fcn.py

```diff
@@ -0,0 +1,355 @@
+            `Conv2D`, `MaxPooling2D` and `Dense` layers are extracted from it
+            to make the custom backbone.
+        model_architecture: str, defines the model architecture based on the
+            implementation details present in the paper. The supported
+            architectures are:
+            1. 'fcn8s', a FCN-8S definition
+            2. 'fcn16s', a FCN-16S definition
+            3. 'fcn32s', a FCN-32S definition
+            Defaults to 'fcn8s'.
+        include_resizing: bool, defines whether to include a `tf.keras.layers.Resizing` layer which defaults to a height and width of value 224. Defaults to `False`
+
+    """
+
+    def __init__(
+        self,
+        classes,
+        backbone="vgg16",
+        model_architecture=None,
+        include_resizing=False,
+        **kwargs
+    ):
+        super(FCN, self).__init__(**kwargs)
+
+        self.resize = include_resizing
+        self.num_classes = classes
+        self.backbone = backbone
+        self.model_architecture = model_architecture
+
+        if self.resize:
+            self.resizing_layer = Resizing(
+                height=224, width=224, interpolation="bilinear"
+            )
+
+    def build(self, input_shape):
+        self.height = input_shape[1]
+        self.width = input_shape[2]
+
+        if isinstance(self.backbone, str):
+
+            if self.backbone == "vgg16":
+                # Assumes `channels_last` format
+                if len(input_shape) == 3:
+                    self.vgg_input_shape = (
+                        input_shape[0],
+                        input_shape[1],
+                        input_shape[2],
+                    )
+                elif len(input_shape) == 4:
+                    self.vgg_input_shape = (
+                        input_shape[1],
+                        input_shape[2],
+                        input_shape[3],
+                    )
+                self.backbone_base_model = VGG16(
+                    include_rescaling=False,
+                    include_top=True,
+                    classes=self.num_classes,
+                    input_shape=self.vgg_input_shape,
+                )
+                self.backbone_name = self.backbone
+            elif self.backbone == "vgg19":
+                self.vgg_input_shape = (input_shape[1], input_shape[2], input_shape[3])
+                self.backbone_base_model = VGG19(
+                    include_rescaling=False,
+                    include_top=True,
+                    classes=self.num_classes,
+                    input_shape=self.vgg_input_shape,
+                )
+                self.backbone_name = self.backbone
+            else:
+                raise ValueError(
+                    "Entered `backbone` argument is not a standard available backbone. Valid options are ['vgg16', 'vgg19']"
+                )
+
+        else:
+            if isinstance(self.backbone, tf.keras.models.Model):
```

Comment:
Our resnet model has `as_backbone` method which returns all stage outputs, is that something we could re-use here?

Reply:
For reference, here's how the usage looks like and how the output maps look like (from `resnet_v2_test.py`):

```
model = resnet_v2.ResNet50V2(
            include_rescaling=False,
            include_top=False,
            classes=2048,
            input_shape=[256, 256, 3],
        )
        backbone_model = model.as_backbone()
        inputs = tf.keras.Input(shape=[256, 256, 3])
        outputs = backbone_model(inputs)
        # Resnet50 backbone has 4 level of features (2 ~ 5)
        self.assertLen(outputs, 4)
        self.assertEquals(list(outputs.keys()), [2, 3, 4, 5])
        self.assertEquals(outputs[2].shape, [None, 64, 64, 256])
        self.assertEquals(outputs[3].shape, [None, 32, 32, 512])
        self.assertEquals(outputs[4].shape, [None, 16, 16, 1024])
        self.assertEquals(outputs[5].shape, [None, 8, 8, 2048])
```

Reply:
Right, it might prove to be quite useful. Let me give it a quick read and see whether it fits perfectly or if there are some changes needed